### DISCUSSION TITLE
When both patch and minor updates available in a combined MR, picking patch

### DISCUSSION THREAD

**Discussion by countergram** [renovatebot/renovate#22153] [Suggest an Idea]
**What Renovate type are you using?**
Self-hosted K8s/GitLab

**Describe the bug**
On a test repo with Express 4.11.1 installed and a wildcard `packagePatterns`, renovate finds a minor update to `4.17.1` as well as a patch update to `4.11.2`, mentions both in the MR description, but chooses `4.11.2` (the older one) rather than the newest version.

**Did you see anything helpful in debug logs?**
I have logs but nothing seems directly relevant to version resolution.

**To Reproduce**

This is my config

```
{
  "assignees": [
    "XXX"
  ],
  "enabled": true,
  "major": {
    "enabled": false
  },
  "packageRules": [
    {
      "groupName": "All Packages",
      "packagePatterns": "*",
      "rangeStrategy": "bump"
    }
  ]
}
```

**Expected behavior**

Dedupe different levels of update to the same package in a way that ends up with the overall latest version.

**Screenshots**

![Screen Shot 2019-10-16 at 9 58 53 AM](https://user-images.githubusercontent.com/117626/66931598-f4899d80-effb-11e9-99ac-ae2c523f159e.png)
![Screen Shot 2019-10-16 at 9 58 44 AM](https://user-images.githubusercontent.com/117626/66931599-f4899d80-effb-11e9-8094-041a4865c143.png)

**Additional context**

**Comment by rarkins** [comment]
Can you create a public github or gitlab repo that reproduces this in a minimal form? It shouldn't make a difference which platform it is, because this is application layer logic.

**Comment by countergram** [comment]
I'll try to create that. Incidentally, it appears this may happen specifically when an existing MR is updated.
